src/DeepCopy/Reflection/ReflectionHelper.php

```diff
@@ -21,16 +21,19 @@ public static function getProperties(\ReflectionClass $ref)
         $propsArr = array();
 
         foreach ($props as $prop) {
-            $f = $prop->getName();
-            $propsArr[$f] = $prop;
+            $propertyName = $prop->getName();
+            $propsArr[$propertyName] = $prop;
         }
 
         if ($parentClass = $ref->getParentClass()) {
             $parentPropsArr = self::getProperties($parentClass);
-            if (count($parentPropsArr) > 0) {
-                $propsArr = array_merge($parentPropsArr, $propsArr);
+            foreach ($propsArr as $key => $property) {
+                $parentPropsArr[$key] = $property;
             }
```

Comment:
The behaviour is different: please add a test where the parent class and the child class has a `private` variable with the same name, and the test verifies that the `ReflectionProperty` is the one of the child instead the one of the parent